essentials/src/collector/mod.rs

```diff
@@ -161,8 +161,8 @@ pub enum CollectorUpdateEvent {
 	NewHead(NewHeadEvent),
 	/// Occurs on a new session
 	NewSession(u32),
-	/// Occurs when collector is disconnected and is about to terminate
-	Termination,
+	/// Occurs when collector is disconnected and is about to terminate, contains error code
+	Termination(i32),
```

Comment:
I think it should be something like 
```rust
enum CollectorTermination {
  Normal,
  Abnormal(i32, String)
}
```

This will distinguish normal from abnormal explicitly and let collector pass extra information about termination reason.